Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/AadhaarVerification.hs

```diff
@@ -0,0 +1,233 @@
+import Data.Text (pack, unpack)
+import qualified Data.Text as T
+import Data.Time.Format.ISO8601 (iso8601Show)
+import qualified Domain.Types.AadhaarVerification.AadhaarOtp as Domain
+import Domain.Types.AadhaarVerification.AadhaarVerification
+import qualified Domain.Types.AadhaarVerification.AadhaarVerification as VDomain
+import qualified Domain.Types.Merchant as DM
+import qualified Domain.Types.Person as Person
+import Environment
+import Kernel.External.Encryption (DbHash, getDbHash)
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Storage.CachedQueries.Merchant as CM
+import qualified Storage.Queries.AadhaarVerification.AadhaarOtp as Query
+import qualified Storage.Queries.AadhaarVerification.AadhaarVerification as QAV
+import qualified Storage.Queries.Person as Person
+import qualified Tools.AadhaarVerification as AadhaarVerification
+import Tools.Error
+
+data VerifyAadhaarOtpReq = VerifyAadhaarOtpReq
+  { otp :: Int,
+    shareCode :: Text
+  }
+  deriving (Show, Generic, ToSchema, ToJSON, FromJSON)
+
+data ImageType = JPG | PNG | UNKNOWN deriving (Generic, Show, Eq)
+
+generateAadhaarOtp ::
+  Bool ->
+  Maybe DM.Merchant ->
+  Id Person.Person ->
+  AadhaarVerification.AadhaarOtpReq ->
+  Flow AadhaarVerification.AadhaarVerificationResp
+generateAadhaarOtp isDashboard mbMerchant personId req = do
+  person <- Person.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  when person.blocked $ throwError (InternalError "Person Account is Blocked")
+  when (person.aadhaarVerified) $ throwError AadhaarAlreadyVerified
+  aadhaarHash <- getDbHash req.aadhaarNumber
+  checkForDuplicacy aadhaarHash
+  whenJust mbMerchant $ \merchant -> do
+    unless (merchant.id == person.merchantId) $ throwError (PersonNotFound personId.getId)
+  let tryKey = makeGenerateOtpTryKey person.id
+  numberOfTries :: Maybe Int <- Redis.safeGet tryKey
+  let tried = fromMaybe 0 numberOfTries
+  merchant <- CM.findById person.merchantId >>= fromMaybeM (MerchantNotFound person.merchantId.getId)
+  unless (isDashboard || tried < merchant.aadhaarVerificationTryLimit) $ throwError (GenerateAadhaarOtpExceedLimit personId.getId)
+  res <- AadhaarVerification.generateAadhaarOtp person.merchantId $ req
+  aadhaarOtpEntity <- mkAadhaarOtp personId res
+  _ <- Query.createForGenerate aadhaarOtpEntity
+  cacheAadhaarVerifyTries personId tried res.transactionId aadhaarHash isDashboard
+  pure res
+
+cacheAadhaarVerifyTries :: Id Person.Person -> Int -> Maybe Text -> DbHash -> Bool -> Flow ()
+cacheAadhaarVerifyTries _ _ Nothing _ _ = return ()
+cacheAadhaarVerifyTries personId tried transactionId aadhaarNumberHash isDashboard = do
+  let key = makeTransactionIdAndAadhaarHashKey personId
+  let tryKey = makeGenerateOtpTryKey personId
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
```

Comment:
this config is one day in prod, do we want to block for a day if exceded?

Reply:
it will a optional   , so  i think now one will try for more  then three times a day

Reply:
but this config we are keeping as 86400, as its used at other places also. we can not update this config in prod, if you want a separate expiryTime for this then add a new config in DB.

Reply:
sure  @VD321   can you do this change

Reply:
Done